src/sas/sasgui/perspectives/fitting/fitpage.py

```diff
@@ -1600,91 +1615,129 @@ def _get_smear_info(self):
         Get the smear info from data.
 
         :return: self.smear_type, self.dq_l and self.dq_r,
-            respectively the type of the smear, dq_min and
-            dq_max for pinhole smear data
-            while dxl and dxw for slit smear
+            respectively the type of the smear, The average <dq/q> radial(p)
+            and <dq/q> theta (s)s for 2D pinhole resolution in % (slit is not
+            currently supported in 2D), (dq/q)_min and (dq/q)_max for 1D pinhole
+            smeared data, again in %, and dxl and/or dxw for slit smeared data
+            given in 1/A and assumed constant.
         """
-        # default
+        # set up defaults
         self.smear_type = None
         self.dq_l = None
         self.dq_r = None
         data = self.data
+        #sanity check - this should only be called if data exits
         if self.data is None:
             return
-        elif self.data.__class__.__name__ == "Data2D" or \
-            self.enable2D:
-            if data.dqx_data is None or data.dqy_data is None:
-                return
-            elif self.current_smearer is not None \
-                and data.dqx_data.any() != 0 \
-                and data.dqx_data.any() != 0:
+        #First check if data is 2D
+        #If so check that data set has smearing info and that none are zero.
+        #Otherwise no smearing can be applied using smear from data (a Gaussian
+        #width of zero will cause a divide by zero error)
+        if self.data.__class__.__name__ == "Data2D":
```

Comment:
That's what I was looking for!! hated those ugly if statements :-)  Sadly need help here.  Was unable to make `isinstance` work, even after importing it and even importing `datainfo` for good measure.  Get the error that `Data2D is undefined global name`
Help?

Reply:
If you imported `data_info`, you would have to refer to the class as `data_info.Data2D`. 

Something like this might work (with the import at the top of the file...).

```
from sas.sascalc.dataloader.data_info import Data2D
if isinstance(self.data, Data2D): [...] 
```

Reply:
Ah  good point - that should fix it, but will need to import both Data2D and Data1D. Though why is it necessary to import from dataloader in the first place when the object clearly knows it is of class Data2(1)D seems silly :-)  Will give it a go. Thanks